modules/gdscript/gdscript_analyzer.cpp

```diff
@@ -313,6 +359,117 @@ Error GDScriptAnalyzer::resolve_inheritance(GDScriptParser::ClassNode *p_class,
+		GDScriptParser::ClassNode *current_class_node = current_data_type->class_type;
+		for (int i = 0; i < p_class->members.size(); i++) {
+			switch (p_class->members[i].type) {
+				case GDScriptParser::ClassNode::Member::VARIABLE:
+					if (has_member_name_conflict_in_script_class(p_class->members[i].variable->identifier->name, current_class_node)) {
+						push_error(vformat(R"(The member "%s" already exists in a parent class.)", p_class->members[i].variable->identifier->name),
+								p_class->members[i].variable);
+						return ERR_PARSE_ERROR;
+					}
+					break;
+				case GDScriptParser::ClassNode::Member::CONSTANT:
+					if (has_member_name_conflict_in_script_class(p_class->members[i].constant->identifier->name, current_class_node)) {
+						push_error(vformat(R"(The member "%s" already exists in a parent class.)", p_class->members[i].constant->identifier->name),
+								p_class->members[i].constant);
+						return ERR_PARSE_ERROR;
+					}
+					break;
+				case GDScriptParser::ClassNode::Member::ENUM:
+					if (has_member_name_conflict_in_script_class(p_class->members[i].m_enum->identifier->name, current_class_node)) {
+						push_error(vformat(R"(The member "%s" already exists in a parent class.)", p_class->members[i].m_enum->identifier->name),
+								p_class->members[i].m_enum);
+						return ERR_PARSE_ERROR;
+					}
+					break;
+				case GDScriptParser::ClassNode::Member::ENUM_VALUE:
+					if (has_member_name_conflict_in_script_class(p_class->members[i].enum_value.identifier->name, current_class_node)) {
+						push_error(vformat(R"(The member "%s" already exists in a parent class.)", p_class->members[i].enum_value.identifier->name),
+								p_class->members[i].enum_value.custom_value);
+						return ERR_PARSE_ERROR;
+					}
+					break;
+				case GDScriptParser::ClassNode::Member::CLASS:
+					if (has_member_name_conflict_in_script_class(p_class->members[i].m_class->identifier->name, current_class_node)) {
+						push_error(vformat(R"(The member "%s" already exists in a parent class.)", p_class->members[i].m_class->identifier->name),
+								p_class->members[i].m_class);
+						return ERR_PARSE_ERROR;
+					}
+					break;
+				case GDScriptParser::ClassNode::Member::SIGNAL:
+					if (has_member_name_conflict_in_script_class(p_class->members[i].signal->identifier->name, current_class_node)) {
+						push_error(vformat(R"(The member "%s" already exists in a parent class.)", p_class->members[i].signal->identifier->name),
+								p_class->members[i].signal);
+						return ERR_PARSE_ERROR;
+					}
+					break;
+				default:
+					break;
+			}
+		}
+		current_data_type = &current_class_node->base_type;
+	}
+
+	// Check for name collisions in native types
+	StringName native_name = "";
+	if (current_data_type && current_data_type->kind == GDScriptParser::DataType::Kind::NATIVE) {
+		native_name = current_data_type->native_type;
+	}
+
+	if (String(native_name) != "") {
+		for (int i = 0; i < p_class->members.size(); i++) {
```

Comment:
Same about this, it can be done on the same loop in `resolve_class_interface()`. There's not really a need for an extra loop just for this check.